packages/react/src/components/RadioButtonGroup/RadioButtonGroup.tsx

```diff
@@ -166,30 +169,38 @@ const RadioButtonGroup = React.forwardRef(
     }
 
     function getRadioButtons() {
-      const mappedChildren = React.Children.map(children, (radioButton) => {
-        const { value } = (radioButton as ReactElement)?.props ?? undefined;
-
-        const newProps = {
-          name: name,
-          key: value,
-          value: value,
-          onChange: handleOnChange,
-          checked: value === selected,
-          required: required,
-        };
-
-        if (!selected && (radioButton as ReactElement)?.props.checked) {
-          newProps.checked = true;
+      const mappedChildren = React.Children.map(
+        children as ReactElement<RadioButtonProps>,
```

Comment:
`children` is now typed as a `RadioButton` allowing us to inherit the types from `RadioButtonProps`. Previously, all prop values referenced in this function were typed as `any`.

This way, we no longer need to keep asserting the type of `radioButton`.